sync/sync_head.go

```diff
@@ -186,7 +192,59 @@ func (s *Syncer[H]) verify(ctx context.Context, newHead H) (bool, error) {
 			"reason", verErr.Reason)
 	}
 
-	return verErr.SoftFailure, err
+	return false, err
+}
+
+// verifySkipping performs a bifurcated header verification process such that
+// it tries to find a header (or several headers if necessary)
+// between the networkHead and the subjectiveHead such that non-adjacent
+// (or in the worst case adjacent) verification passes and the networkHead
+// can be verified as a valid sync target against the syncer's subjectiveHead.
+//
+// When such candidates cannot be found [NewValidatorSetCantBeTrustedError] will be returned.
+func (s *Syncer[H]) verifySkipping(ctx context.Context, subjHead, networkHeader H) error {
```

Comment:
Comment uses `networkHead` while var name is `networkHeader`. Lets stick `networkHead`

Reply:
Nice catch, fixed!